README.md

```diff
@@ -677,10 +680,96 @@ player.ready(function() {
   });
 });
 ```
+Note: any registered `onRequest` hooks, are called _after_ the `beforeRequest` function, so xhr
+options modified by this function may be further modified by these hooks.
 
 For information on the type of options that you can modify see the
 documentation at [https://github.com/Raynos/xhr](https://github.com/Raynos/xhr).
 
+#### vhs.onRequest
+Type: `function`
+
+The `onRequest(callback)` function takes a `callback` function that will pass the xhr `request` 
+Object to that callback. These callbacks are called in the order registered and act as pre-request 
+hooks for modifying the xhr `request` Object prior to making a request. These hooks can be set at 
+the global or player level, where the player level hooks will override global.
+
+
+Example
+```javascript
+// Global
+const globalRequestHook = (request) => {
+  const requestUrl = new URL(request.uri);
+  requestUrl.searchParams.set('foo', 'bar');
+  request.uri = decodeURIComponent(requestUrl.href);
```

Comment:
Does this need to be passed to `decodeURIComponent`? 🤔

Reply:
Good catch, it does not, just tested it. Got a little copy paste happy this morning for the docs. 😅

Reply:
Removed